packages/core/admin/ee/server/services/seat-enforcement.js

```diff
@@ -1,161 +1,114 @@
+ */
+const getDisabledUserList = async () => {
+  return strapi.store.get({ type: 'ee', key: 'disabled_users' });
+};
 
-  const disabledUsers = JSON.parse(data.value);
+const enableMaximumUserCount = async (numberOfUsersToEnable) => {
+  const disabledUsers = await getDisabledUserList();
+  const orderedDisabledUsers = reverse(disabledUsers);
 
-  const usersToEnable = take(disabledUsers, numberOfUsersToEnable);
+  const usersToEnable = take(numberOfUsersToEnable, orderedDisabledUsers);
 
-  for await (const user of usersToEnable) {
-    try {
-      await strapi.db.query('admin::user').update({
-        where: { id: user.id },
-        data: { isActive: true },
-      });
-    } catch (error) {
-      return;
-    }
-  }
+  await strapi.db.query('admin::user').updateMany({
+    where: { id: map(prop('id'), usersToEnable) },
+    data: { isActive: true },
+  });
 
-  const remainingDisabledUsers = drop(disabledUsers, numberOfUsersToEnable);
+  const remainingDisabledUsers = drop(numberOfUsersToEnable, orderedDisabledUsers);
 
-  await strapi.db.query('strapi::ee-store').update({
-    where: { id: data.id },
-    data: { value: JSON.stringify(remainingDisabledUsers) },
+  await strapi.store.set({
+    type: 'ee',
+    key: 'disabled_users',
+    value: remainingDisabledUsers,
   });
 };
 
-const calculateAdminSeatDifference = async (seatsAllowedByLicense) => {
-  const currentAdminSeats = await getService('user').getCurrentActiveUserCount();
-  return currentAdminSeats - seatsAllowedByLicense;
-};
-
 const disableUsersAboveLicenseLimit = async (numberOfUsersToDisable) => {
-  const users = await strapi.db.query('admin::user').findMany({
-    where: { isActive: 'true' },
+  const currentlyDisabledUsers = (await getDisabledUserList()) ?? [];
+
+  const usersToDisable = [];
+  const nonSuperAdminUsersToDisable = await strapi.db.query('admin::user').findMany({
+    where: {
+      isActive: true,
+      roles: {
+        code: { $ne: SUPER_ADMIN_CODE },
+      },
+    },
     orderBy: { createdAt: 'DESC' },
-    populate: { roles: { select: ['id'] } },
+    limit: numberOfUsersToDisable,
   });
 
-  const usersToDisable = take(users, numberOfUsersToDisable);
-
-  for await (const user of usersToDisable) {
-    try {
-      await strapi.db.query('admin::user').update({
-        where: { id: user.id },
-        data: {
-          isActive: false,
-        },
-      });
-      user.isActive = false;
-    } catch (error) {
-      return;
-    }
-  }
-
-  const data = await strapi.db.query('strapi::ee-store').findOne({
-    where: { key: 'ee_disabled_users' },
-  });
+  usersToDisable.push(...nonSuperAdminUsersToDisable);
 
-  if (data) {
-    await strapi.db.query('strapi::ee-store').update({
-      where: { id: data.id },
-      data: { value: JSON.stringify(usersToDisable) },
+  if (nonSuperAdminUsersToDisable.length < numberOfUsersToDisable) {
+    const superAdminUsersToDisable = await strapi.db.query('admin::user').findMany({
```

Comment:
Could we ever reach a scenario where we've disabled all the users?

Reply:
it could if you have 0 seats but that's not really an option

Reply:
That's what I thought, just wanted to be sure :+1: